src/services/CleanupService.ts

```diff
@@ -404,110 +406,218 @@ export const clearLocalstorageForThisDomain = async (
   }
 };
 
-/** This will use the browsingData's hostname/origin attribute to delete any extra browsing data */
-export const otherBrowsingDataCleanup = async (
+export const clearSiteDataForThisDomain = async (
   state: State,
-  domains: string[],
+  siteData: string,
+  hostname: string,
 ): Promise<boolean> => {
+  if (hostname.trim() === '') return false;
   const debug = getSetting(state, 'debugMode') as boolean;
-  if (getSetting(state, 'localstorageCleanup')) {
-    if (
-      isFirefoxNotAndroid(state.cache) &&
-      state.cache.browserVersion >= '58'
-    ) {
-      const cleanList: string[] = [];
-      for (const domain of domains) {
-        for (const d of prepareCleanupDomains(domain)) {
-          if (d !== '' && d.trim() !== '') cleanList.push(d);
-        }
-      }
-      if (cleanList.length === 0) return false;
-      const hostnames = [...new Set(cleanList)];
-      cadLog(
-        {
-          msg:
-            'CleanupService.otherBrowsingDataCleanup: Hostnames sent to Firefox LocalStorage Cleanup:',
-          x: hostnames,
-        },
-        debug,
-      );
-      await browser.browsingData
-        .removeLocalStorage({
-          hostnames,
-        })
-        .catch((e) => {
-          cadLog(
-            {
-              msg:
-                'CleanupService.otherBrowsingDataCleanup: removeLocalStorage returned an error:',
-              type: 'error',
-              x: e,
-            },
-            debug,
-          );
-          throw e;
-        });
-      return true;
+  cadLog(
+    {
+      msg: `CleanupService.clearSiteDataForThisDomain: Received ${siteData} clean request for ${hostname}.`,
+    },
+    debug,
+  );
+  const domains = prepareCleanupDomains(hostname, isChrome(state.cache));
+  if (siteData === 'All') {
+    for (const sd of [
+      'cache',
+      'cookies',
+      'indexedDB',
+      'localStorage',
+      'pluginData',
+      'serviceWorkers',
+    ]) {
+      await removeSiteData(sd, isChrome(state.cache), domains, debug);
     }
-    if (isChrome(state.cache)) {
-      const cleanList: string[] = [];
-      for (const domain of domains) {
-        for (const d of prepareCleanupDomains(domain)) {
-          if (d !== '' && d.trim() !== '') {
-            cleanList.push(`http://${d}`);
-            cleanList.push(`https://${d}`);
-          }
-        }
-      }
-      if (cleanList.length === 0) return false;
-      const origins = [...new Set(cleanList)];
-      cadLog(
-        {
-          msg: `CleanupService.otherBrowsingDataCleanup: Origins sent to Chrome LocalStorage Cleanup`,
-          x: origins,
-        },
+  } else {
+    await removeSiteData(
+      `${siteData[0].toLowerCase()}${siteData.slice(1)}`,
+      isChrome(state.cache),
+      domains,
+      debug,
+    );
+  }
+  return true;
+};
+
+export const removeSiteData = async (
+  siteData: string,
+  isChrome: boolean,
```

Comment:
Hmm, maybe not making it specific here would be best (not boolean). Something like a browser enum would be more futureproof for future browsers. Because otherwise, your options here are Chrome and Not Chrome.

Then in the body you can do a switch statement with the different implementations.

Reply:
Implemented in commit ac90490